### DISCUSSION TITLE
Is there any plans to change the way that the packages are stored (like pnpm)?

### DISCUSSION THREAD

**Discussion by codermarcos** [npm/feedback#1075] [CLI]
I would like to know if there is a plan to change the way that packages are stored (like pnpm).

I know about the limitations due to compatibility as I've read in this [discussion](https://github.com/npm/feedback/discussions/341) but It didn't look that It would be solved as soon.

However the number of download's [pnpm](https://www.npmjs.com/package/pnpm) is increasing every day so there is a large amount of people looking for this feature.

Would it not be possible to create a configuration for it, to indicate whether that the library can be stored separately from project or it should be stored one for each project?

**Comment by ljharb** [comment]
npm already has an "isolated" layout mode that is designed to be pnpm-like.
